cmd/controller/metrics.go

```diff
@@ -0,0 +1,71 @@
+package main
+
+import (
+	"github.com/go-kit/kit/metrics"
+	"github.com/go-kit/kit/metrics/prometheus"
+	prom "github.com/prometheus/client_golang/prometheus"
+)
+
+// MachineControllerMetrics is a struct of all metrics used in
+// the machine controller.
+type MachineControllerMetrics struct {
+	Machines            metrics.Gauge
+	Nodes               metrics.Gauge
+	Workers             metrics.Gauge
+	Errors              metrics.Counter
+	ControllerOperation metrics.Histogram
+	NodeJoinDuration    metrics.Histogram
+}
+
+// NewMachineControllerMetrics creates new MachineControllerMetrics
+// with default values initialized, so metrics always show up.
+func NewMachineControllerMetrics() *MachineControllerMetrics {
+	namespace := "machine"
+	subsystem := "controller"
+
+	cm := &MachineControllerMetrics{
+		Machines: prometheus.NewGaugeFrom(prom.GaugeOpts{
+			Namespace: namespace,
+			Subsystem: subsystem,
+			Name:      "machines",
+			Help:      "The number of currently managed machines",
```

Comment:
Maybe omit the `currently managed` as there is no way a machine can exist but not be managed?